### PR TITLE
Small bug fixes on Import Dataset dialog

### LINKED ISSUE #6229: Another bug in the File > Open from File dialogue
**Describe the bug**
a) I use File Open and then open 6 Excel sheets from the library - in the Instat Experimental file.  This is fine
b) I return to the dialogue and now open a single file Chipata1.dly
c) It refuses and the second line shows it has not changed the saving line.  It is still trying to save the 6 data frames from the previous command.

Here is the error message:
image
 
I can reproduce the problem just from the library data sets.  Also also show a second problem!

a) In Open from library use step a above - fine
b) Return to the previous command (Open)
c) Notice the second bug to me anyway.  It has remembered the previous sheets, in the Excel file, but Ok is not enabled.  Either it shouldn't "remember" the sheets, or the Ok should be enabled.  But that is secondary here.
d) Go to library again and choose Climatic > Nigeria and samaru56.csv file.  Notice the dialogue hasn't changed, so it doesn't give me the option of a data frame name.
e) Ok gives the error.

### PR DESCRIPTION
**Pull request by N-thony**
fixes #6229 
@africanmathsinitiative/developers @rdstern this is ready for review. Thanks

### CODE REVIEW

File: instat/dlgImportDataset.vb [hunk 1]
```diff
@@ -636,15 +638,22 @@ Public Class dlgImportDataset
...
+                ucrSaveFile.SetAssignToBooleans(bTempDataFrameList:=True)
+                If clbSheets.CheckedItems.Count > 1 Then
+                    ucrSaveFile.Hide()
+                ElseIf clbSheets.CheckedItems.Count = 1 Then
+                    ucrSaveFile.Show()
+                    ucrSaveFile.SetName(dctSelectedExcelSheets.Values.First(), bSilent:=True)
+                ElseIf clbSheets.CheckedItems.Count = 0 Then
+                    ucrSaveFile.Show()
+                    ucrSaveFile.SetName(frmMain.clsRLink.MakeValidText(strFileName), bSilent:=True)
+                End If
```
Comment (lloyddewit) [inline instat/dlgImportDataset.vb:651]: Replace with a select/case statement?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by N-thony** [comment]
@lloyddewit  thanks for the comment above. I think have resolved it.

**Review by lloyddewit — DISMISSED** [review]

**Comment by lloyddewit** [comment]
@rdstern please could you test? thanks

**Review by rdstern — APPROVED** [review]

**Comment by shadrackkibet** [comment]
I tested and found some minor design issues that should be easy to fix here instead of another PR. See the below screenshot.

- There is a control that overlaps with preview when loading the `.dly` file
- Remove unnecessary space between the label and input in the save control

image

**Comment by rdstern** [comment]
@shadrackkibet sounds good

**Comment by N-thony** [comment]
@shadrackkibet thanks for your comments. I think have resolved them? could you test?

**Comment by shadrackkibet** [comment]
@N-thony can you edit this PR title so that it contains the details of the bug fixed and the changes made? Check the message below from Danny sent to your mailbox.

>1. Pull Request Titles
The PR title will be used in the release notes. Please make sure the title of your PR is descriptive of the change you are making (but not too long). Also, try to make it less technical as possible. These will be read by users, so imagine a user without programming knowledge reading your PR title. Will they understand what this actually changes in the software? Sometimes it's unavoidable that the title will be technical, but for most dialog changes, we should be able to make them clear to users. In general, include the dialog name in the title if the PR is on a specific dialog.

>Examples
Good: Corrected consecutive wet days calculation on rainfall quality control
Good: Cumulative/Exceedance Graph: Added points option
Good: Small bug fixes on NetCDF dialog
Bad: issue #8888
Bad: Bug fix
Bad: dlgMakeDate

>The PR title can be edited after its created so you can improve it if you think of a more appropriate title. Reviewers should also review the PR title and check it's a sensible description.

**Comment by shadrackkibet** [comment]
Another problem is that the new data frame name is not retained on reopen. It is always reset to the default name.

**Comment by N-thony** [comment]
> Another problem is that the new data frame name is not retained on reopen. It is always reset to the default name.

@shadrackkibet thanks for your comment, would you come again. I have tried this and the new data frame name is retained on reopening.

**Comment by shadrackkibet** [comment]
> > Another problem is that the new data frame name is not retained on reopen. It is always reset to the default name.
> 
> @shadrackkibet thanks for your comment, would you come again. I have tried this and the new data frame name is retained on reopening.

- Open a file say `.csv` from your directory
- Change the **New Data frame name** in the dialog to say "myData"
- Click Ok / Close the dialog 
- Now reopen the dialog

Notice that the name ("myData") is lost.

**Comment by N-thony** [comment]
> > > Another problem is that the new data frame name is not retained on reopen. It is always reset to the default name.
> > 
> > 
> > @shadrackkibet thanks for your comment, would you come again. I have tried this and the new data frame name is retained on reopening.
> 
> * Open a file say `.csv` from your directory
> * Change the **New Data frame name** in the dialog to say "myData"
> * Click Ok / Close the dialog
> * Now reopen the dialog
> 
> Notice that the name ("myData") is lost.

@shadrackkibet I think it is fine since the New Data frame name always by default will be the file name. If you click on a file, it sets the name of the file.

**Comment by lloyddewit** [comment]
> 
> @shadrackkibet I think it is fine since the New Data frame name always by default will be the file name. If you click on a file, it sets the name of the file.

@N-thony As a general principle, if the user closes a dialog, and then reopens it later, the dialog should return to the state it was in when it was closed. Therefore I think @shadrackkibet 's comment from 08 March is still open. Is it difficult to fix?

**Comment by N-thony** [comment]
@lloyddewit thanks for your comment. I think have resolved it.

**Comment by lloyddewit** [comment]
> 
> 
> > @shadrackkibet I think it is fine since the New Data frame name always by default will be the file name. If you click on a file, it sets the name of the file.
> 
> @N-thony As a general principle, if the user closes a dialog, and then reopens it later, the dialog should return to the state it was in when it was closed. Therefore I think @shadrackkibet 's comment from 08 March is still open. Is it difficult to fix?

@N-thony I tested and this now seems to be resolved - thanks

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern please could you retest? thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
The problem is changed but partly still remains.
I opened an Excel file from the library as described before - it worked fine.
Then I tried to open a new file and it gave me the provisional name of the previous one.  This is an occasion when the name freom before should be updated, rather than just remembered.

**Comment by N-thony** [comment]
@rdstern thanks for your comment above. I think have resolved it. Please could you test?

**Review by rdstern — APPROVED** [review]

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**